schemas/iso19115-3.2018/src/main/plugin/iso19115-3.2018/convert/ogcwxs-info-injection.xsl

```diff
@@ -792,65 +791,96 @@
               or . = 'restricted'
+              or . = 'otherRestrictions'
+              or . = 'unrestricted'
+              or . = 'licenceUnrestricted'
+              or . = 'licenceEndUser'
+              or . = 'licenceDistributor'
+              or . = 'private'
+              or . = 'statutory'
+              or . = 'confidential'
+              or . = 'SBU'
+              or . = 'in-confidence'
               ">
             <mco:accessConstraints>
+              <xsl:attribute name="gco:nilReason" select="$nilReasonValue"/>
               <mco:MD_RestrictionCode
-                codeList="http://standards.iso.org/iso/19139/resources/gmxCodelists.xml#MD_RestrictionCode"
+                codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xml#MD_RestrictionCode"
                 codeListValue="{.}"/>
             </mco:accessConstraints>
           </xsl:when>
+          <!-- Second option: if there is an entry in the template record use that -->
+          <xsl:when test="$useLimitation != ''">
+            <xsl:apply-templates mode="copy" select="$legalConstraints"/>
+          </xsl:when>
+          <!-- Third option: if GetCaps response has "<AccessConstraints>" set to "NONE" then output 'no conditions apply' -->
           <xsl:when test="lower-case(.) = 'none'">
             <mco:accessConstraints>
               <mco:MD_RestrictionCode
-                codeList="http://standards.iso.org/iso/19139/resources/gmxCodelists.xml#MD_RestrictionCode"
+                codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xml#MD_RestrictionCode"
                 codeListValue="otherRestrictions"/>
             </mco:accessConstraints>
             <mco:otherConstraints>
+              <xsl:attribute name="gco:nilReason" select="$nilReasonValue"/>
               <gco:CharacterString>no conditions apply</gco:CharacterString>
             </mco:otherConstraints>
           </xsl:when>
+          <!-- Else copy the value from GetCaps "<AccessConstraints>"  -->
           <xsl:otherwise>
             <mco:accessConstraints>
               <mco:MD_RestrictionCode
-                codeList="http://standards.iso.org/iso/19139/resources/gmxCodelists.xml#MD_RestrictionCode"
+                codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xml#MD_RestrictionCode"
                 codeListValue="otherRestrictions"/>
             </mco:accessConstraints>
             <mco:otherConstraints>
+              <xsl:attribute name="gco:nilReason" select="$nilReasonValue"/>
               <gco:CharacterString>
                 <xsl:value-of select="."/>
               </gco:CharacterString>
             </mco:otherConstraints>
           </xsl:otherwise>
         </xsl:choose>
       </mco:MD_LegalConstraints>
+      <!-- Copy security constraints & general constraints from the template record -->
+      <xsl:apply-templates mode="copy" select="$securityConstraints"/>
```

Comment:
This will create an element 
```xml
<mri:resourceConstraints>
<mco:MD_LegalConstraints>
...
</mco:MD_LegalConstraints>
<mco:MD_SecurityConstraints ...
<mco:MD_Constraints ...
```
which is invalid I think.
You've to pack them in separated `<mri:resourceConstraints>`? Maybe change 
`$resConstraints/mco:MD_SecurityConstraints` by `$resConstraints[mco:MD_SecurityConstraints]` for the securityConstraints variable?

Reply:
I cannot relate to this, but I have fixed it below.

Now whether the input packages each one in '<mri:resourceConstraints>..</mri:resourceConstraints>' or not, it will always package the output in '<mri:resourceConstraints>..</mri:resourceConstraints>'